### PR TITLE
#7464 Update add bookmark toast

### PR DESCRIPTION
**Pull request by electricRGB**
Hello. This is "waz happenin" is this fun PR:
- Consolidates where the "Bookmark added" toast is shown from (should be one place instead of three)
- Updates toast to use button style as per spec
- Tapping Edit button opens the existing edit bookmark panel. This is NOT in the Library Panel, but a separate view
- Adds telemetry to new edit button & to existing edit bookmark action in bookmark panel edit bookmark table view

### CODE REVIEW

File: Client/metrics.yaml [hunk 1]
```diff
@@ -223,6 +223,23 @@ bookmarks:
...
+      be edited from:
+      * Add bookmark toast Edit button
+      * Bookmarks panel edit bookmarks view
+    labels:
+      - add-bookmark-toast
+      - bookmarks-panel
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/7464
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/8671
```
Comment (dnarcese) [inline Client/metrics.yaml:239]:
```suggestion
      - https://github.com/mozilla-mobile/firefox-ios/pull/8675
```

**Review by dnarcese — COMMENTED** [review]
The font is different than before and different from the "edit". Is that intentional?

File: Client/metrics.yaml (hunk 1 above)
Comment (electricRGB) [reply Client/metrics.yaml:239]: I always forget to update this. I also have to make the data sheet. totally forgot about that.

**Comment by electricRGB** [comment]
No. I used the toast we had before, as I'd clarified with Bram that we weren't creating a new one, just using the one we had.

**Comment by electricRGB** [comment]
# Request for data collection review form 
1. What questions will you answer with this data?
	* How many users click the Edit button on the add bookmark toast
	* How many users edit a bookmark from the bookmark panel

2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
	* It will provide information essential for understanding importance of the various items, what's important to users, and prioritize future work based on it

3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
	* No other way to collect this data

4. Can current instrumentation answer these questions?
	* No, we didn't have events for these items 

5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement Description | Data Collection Category | Tracking Bug # |
| --- | ----- | ----- |
Counts the number of times a user edits a bookmark from either the add bookmark toast or from the bookmarks panel edit view | 1 | https://github.com/mozilla-mobile/firefox-ios/pull/8675

6. How long will this data be collected? Choose one of the following:
	* I want this data to be collected for 6 months initially (potentially renewable).

7. What populations will you measure?
All countries and channels that have data collection on.

8. If this data collection is default on, what is the opt-out mechanism for users?
	* Toggle switch for 'Send Usage Data' in the app settings.

9. Please provide a general description of how you will analyze this data.
	* By monitoring the count of how often users are performing these actions.

10. Where do you intend to share the results of your analysis?
	* With the mobile team.

11. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
	* No.
	
Documentation has been updated in this PR and will be reflected in our [metrics.md](https://github.com/mozilla-mobile/firefox-ios/blob/main/Docs/metrics.md) file

**Comment by dnarcese** [comment]
Collection Category should be 2, and tracking bug #7464, otherwise r+

# Data Review Form

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, telemetry is documented in `docs/metrics.md`

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, Firefox iOS provides data controls in the app settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

This should be re-reviewed w/ Ashley Thomas every 6mo

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Type 2

5) Is the data collection request for default-on or default-off?

Default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

**Review by dnarcese — APPROVED** [review]

**Pull request merged**